crates/framework/mos-stdlib/sources/object.move

```diff
@@ -0,0 +1,160 @@
+    /// This is the only way to create `UID`s.
+    public fun new(ctx: &mut TxContext): UID {
+        UID {
+            id: ID { bytes: tx_context::new_object(ctx) },
+        }
+    }
+
+    /// Delete the object and it's `UID`. This is the only way to eliminate a `UID`.
+    // This exists to inform Sui of object deletions. When an object
+    // gets unpacked, the programmer will have to do something with its
+    // `UID`. The implementation of this function emits a deleted
+    // system event so Sui knows to process the object deletion
+    public fun delete(id: UID) {
+        let UID { id: ID { bytes } } = id;
+        delete_impl(bytes)
+    }
+
+    /// Get the underlying `ID` of `obj`
+    public fun id<T: key>(obj: &T): ID {
+        borrow_uid(obj).id
+    }
+
+    /// Borrow the underlying `ID` of `obj`
+    public fun borrow_id<T: key>(obj: &T): &ID {
+        &borrow_uid(obj).id
+    }
+
+    /// Get the raw bytes for the underlying `ID` of `obj`
+    public fun id_bytes<T: key>(obj: &T): vector<u8> {
+        bcs::to_bytes(&borrow_uid(obj).id)
+    }
+
+    /// Get the inner bytes for the underlying `ID` of `obj`
+    public fun id_address<T: key>(obj: &T): address {
+        borrow_uid(obj).id.bytes
+    }
+
+    /// Get the `UID` for `obj`.
+    /// Safe because Sui has an extra bytecode verifier pass that forces every struct with
+    /// the `key` ability to have a distinguished `UID` field.
+    /// Cannot be made public as the access to `UID` for a given object must be privileged, and
+    /// restrictable in the object's module.
+    native fun borrow_uid<T: key>(obj: &T): &UID;
+
+    // /// Generate a new UID specifically used for creating a UID from a hash
+    // public(friend) fun new_uid_from_hash(bytes: address): UID {
+    //     record_new_uid(bytes);
+    //     UID { id: ID { bytes } }
+    // }
+
+    // helper for delete
+    native fun delete_impl(id: address);
+
+    //TODO we need this?
+    // marks newly created UIDs from hash
+    //native fun record_new_uid(id: address);
+
+    // === transfer functions ===
+    //https://github.com/MystenLabs/sui/blob/598f106ef5fbdfbe1b644236f0caf46c94f4d1b7/crates/sui-framework/sources/transfer.move#L92
+
+   
+    /// Transfer ownership of `obj` to `recipient`. `obj` must have the
+    /// `key` attribute, which (in turn) ensures that `obj` has a globally
+    /// unique ID.
+    public fun transfer<T: key>(obj: T, recipient: address) {
+        // TODO: emit event
+        transfer_internal(obj, recipient)
+    }
+
+    /// Freeze `obj`. After freezing `obj` becomes immutable and can no
+    /// longer be transferred or mutated.
+    public native fun freeze_object<T: key>(obj: T);
+
+    /// Turn the given object into a mutable shared object that everyone
+    /// can access and mutate. This is irreversible, i.e. once an object
+    /// is shared, it will stay shared forever.
+    public native fun share_object<T: key>(obj: T);
```

Comment:
Object in Sui are owned object default, so objects have transfer, freeze, share operate.
In Rooch objects are share object default, these functions should not reserve。